### DISCUSSION TITLE
Registry key check for WV2 installation

### DISCUSSION THREAD

**Discussion by amcriste** [MicrosoftEdge/WebView2Feedback#3951] [Q&A]
Hi,

During the last 2 updates for WV2 Runtime, I noticed that I get a registry key in HKLM with pv = 0.0.0.0, while in HKCU I find the valid one:
HKLM: pv=0.0.0.0
HKCU: location, name, SilentUninstall populated, pv=100.0.1185.29

At first, I was thinking maybe this was a consequence of various testing done for the project I work on and I deleted the key, but behavior reappeared after update to 100.0.1185.29. 

Is the above expected?

[Documentation](https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/distribution) states that one of these 2 
```
HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\EdgeUpdate\Clients\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}
HKEY_CURRENT_USER\Software\Microsoft\EdgeUpdate\Clients\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}
```
should be present and defined. Only null or empty strings are mentioned for pv value as indication of WV2 not installed. 

Could you please give instructions on what would be the correct way to check the above? Should we check always in both place and account for invalid? Does one hive have priority over the other? Is pv="0.0.0.0" an issue on your side?

Thank you,
Amelia

**Comment by champnic** [comment]
@amcriste Thanks for the question Amelia. I'll reach out to our installer team to see if there's a bug or if we need to update our docs.

Was the evergreen runtime already installed in HKLM and then changed to HKCU after an update?

**Comment by amcriste** [comment]
@champnic no, the installation was in HKCU. I did however uninstall & reinstall several times for testing purposes a few weeks before the issue appeared, but I don't think is related. 
As far as I have seen, if the installation was in HKLM for a normal user (made before CU was supported), it wasn't moved with updates (at least when I checked in early in the year)

**Comment by champnic** [comment]
@amcriste I've confirmed that the 0.0.0.0 is By Design. We'll update our docs to indicate this.

**Comment by luclt90** [comment]
I just checked it again and I realized that:

1. If you install WebView2 Runtime manually (download into local and setup by double click on MicrosoftEdgeWebview2Setup.exe), then the key is available at HKEY_CURRENT_USER\SOFTWARE\Microsoft\EdgeUpdate\Clients{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}
2. If you install WebView2 Runtime by command. In my case, I installed it from NSIS file setup, then the key is available at HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\EdgeUpdate\Clients{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}:
```
Function installWebView2
	ClearErrors
	ReadRegStr $0 HKLM "SOFTWARE\WOW6432Node\Microsoft\EdgeUpdate\Clients\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}" "pv"
	${If} ${Errors} 
	${OrIf} $0 == ""
		DetailPrint "Dowloading: WebView2 Runtime"		
		InetLoad::load "http://go.microsoft.com/fwlink/p/?LinkId=2124703" "$TEMP\MicrosoftEdgeWebview2Setup.exe"
        Pop $R0
		StrCmp $R0 "OK" OK
			MessageBox MB_OK|MB_ICONEXCLAMATION "Failed to download WebView2 Runtime. ${PRODUCT_NAME} will continue to install, but the Map in ${PRODUCT_NAME} will not run properly without WebView2 Runtime. You must install WebView2 Runtime manually."
			Goto done_installWebView2_function
        OK:
		DetailPrint "Installing: WebView2 Runtime"
		ExecWait '"$TEMP\MicrosoftEdgeWebview2Setup.exe" /silent /install'
	${EndIf}
	done_installWebView2_function:
FunctionEnd
```
Just share my experience. Maybe can help
